ivy_models/clip/layers.py

```diff
@@ -0,0 +1,411 @@
+from typing import Union
+
+import ivy
+from ivy.stateful.initializers import RandomNormal
+
+
+class Identity(ivy.Module):
```

Comment:
seems like a general layer, I would suggest adding it to ivy stateful API

Reply:
Yes it is. Same for the `Embedding` layer. There is a reported issue with the `ivy.embedding` function, so I'll look into it to fix it and then add the `Embedding` and `Identity` layers in the stateful API. I left a `# TODO` in the code to refactor it when the PR for adding both layers to `ivy` is merged.